libs/asset-viewer/src/lib/services/asset-search.service.ts

```diff
@@ -1,90 +1,49 @@
-            .pipe(
-                map(flow(SearchAssetResultClientDecoder.decode, E.mapLeft(decodeError))),
-                OE.catchErrorW(httpErrorResponseError),
-                map(RD.fromEither),
-                startWith(RD.pending),
-            );
-    }
-
-    public searchByPolygon(polygon: LV95[]): ORD.ObservableRemoteData<ApiError, SearchAssetResultClient> {
-        const qs = AssetSearchParamsToQueryString.encode({ filterKind: 'polygon', polygon, searchText: O.none });
-        return this._httpClient
-            .get('/api/asset/?' + qs)
-            .pipe(
-                map(flow(SearchAssetResultClientDecoder.decode, E.mapLeft(decodeError))),
-                OE.catchErrorW(httpErrorResponseError),
-                map(RD.fromEither),
-                startWith(RD.pending),
-            );
-    }
-
-    public searchByPolygonRefineBySearchText(
-        polygon: LV95[],
-        searchText: string,
-    ): ORD.ObservableRemoteData<ApiError, SearchAssetResultClient> {
-        const qs = AssetSearchParamsToQueryString.encode({
-            filterKind: 'polygon',
-            polygon,
-            searchText: O.some(searchText),
-        });
-        return this._httpClient
-            .get('/api/asset/?' + qs)
-            .pipe(
-                map(flow(SearchAssetResultClientDecoder.decode, E.mapLeft(decodeError))),
-                OE.catchErrorW(httpErrorResponseError),
-                map(RD.fromEither),
-                startWith(RD.pending),
-            );
-    }
-
-    public loadReferenceData(): ORD.ObservableRemoteData<ApiError, ReferenceData> {
-        return this._httpClient
-            .get('/api/reference-data')
-            .pipe(
-                map(flow(ReferenceData.decode, E.mapLeft(decodeError))),
-                OE.catchErrorW(httpErrorResponseError),
-                map(RD.fromEither),
-                startWith(RD.pending),
-            );
-    }
-
-    public loadAssetDetailData(assetId: number): ORD.ObservableRemoteData<ApiError, AssetDetail> {
-        return this._httpClient
-            .get(`/api/asset-detail/${assetId}`)
-            .pipe(
-                map(flow(AssetDetail.decode, E.mapLeft(decodeError))),
-                OE.catchErrorW(httpErrorResponseError),
-                map(RD.fromEither),
-                startWith(RD.pending),
-            );
-    }
+  constructor(private _httpClient: HttpClient, private store: Store<AppState>) {
+  }
+
+  public search(searchQuery: AssetSearchQuery): Observable<AssetSearchResult> {
+    return this._httpClient
+      .post('/api/assets/search?limit=10000', searchQuery)
+      .pipe(
+        map((res) => plainToInstance(AssetSearchResultDTO, res)),
+        tap((result) => {
+          result.data = result.data.map((asset) => (AssetEditDetail.decode(asset) as E.Right<AssetEditDetail>).right);
+        }),
+      );
+  }
+
+  public loadAssetDetailDataNew(assetId: number): Observable<AssetEditDetail> {
```

Comment:
Remove unnecessary suffix.
```suggestion
  public loadAssetDetailData(assetId: number): Observable<AssetEditDetail> {
```